qcodes/instrument_drivers/Keysight/keysight_e4980a.py

```diff
@@ -433,6 +435,31 @@ def _get_current_level(self) -> float:
 
         return float(i_level)
 
+    def _is_signal_mode_voltage_on_driver_initialization(self) -> bool:
+        """
+        Checks if signal is set with voltage_level param at instrument driver
+        initialization
+        """
+        assert self.signal_mode() is None
+        try:
+            self.voltage_level()
+            return True
+        except VisaIOError:
+            return False
+
+    def _set_signal_mode_on_driver_initialization(self) -> None:
+        """
+        Sets signal mode on driver initialization
+        """
+        if self._is_signal_mode_voltage_on_driver_initialization():
+            self.signal_mode("Voltage")
+            self.voltage_level.snapshot_exclude = False
+            self.current_level.snapshot_exclude = True
+        else:
+            self.signal_mode("Current")
+            self.voltage_level.snapshot_exclude = True
+            self.current_level.snapshot_exclude = False
```

Comment:
Don't we need to change which parameters are excluded when we change mode in `_set_voltage_level` and `_set_current_level` too

Reply:
Yes, we do. Will set these there as well.